src/views/catalog/customize/components/CustomizeForms/useCustomizeFormSubmit.ts

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+
+import { getTemplateVirtualMachineObject } from '../../../utils/templateGetters';
+import { useWizardVMContext } from '../../../utils/WizardVMContext';
+import { processTemplate } from '../../utils';
+
+type useCustomizeFormSubmitType = [
+  onSubmit: (event: any) => Promise<void>,
+  loaded: boolean,
+  error: any,
+];
+
+export const useCustomizeFormSubmit = (template: V1Template): useCustomizeFormSubmitType => {
+  const { ns } = useParams<{ ns: string }>();
+  const history = useHistory();
+  const [templateLoaded, setTemplateLoaded] = React.useState(true);
+  const [templateError, setTemplateError] = React.useState<any>();
+
+  const { updateVM, loaded: vmLoaded, error: vmError } = useWizardVMContext();
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    setTemplateLoaded(false);
+    try {
+      const formData = new FormData(event.currentTarget as HTMLFormElement);
+
+      const processedTemplate = await processTemplate(template, ns, formData);
+
+      const vm = getTemplateVirtualMachineObject(processedTemplate);
+      updateVM(vm);
```

Comment:
we should `await` updateVM before `history.push`
Are you displaying the error that comes from `useWizardVMContext` somewhere?

Reply:
Okay, thanks. No, I'll do a different pr to show the errors.